### ISSUE TITLE
SP5 assemblies are not signed

### ISSUE THREAD

**Issue by mlessmann** [ScottPlot/ScottPlot#3235]
**Issue:**
The assemblies of ScottPlot 5 are not signed. Consumers that are signed therefore get the following error during build:
`error CS8002: Referenced assembly 'ScottPlot, Version=5.0.18.0, Culture=neutral, PublicKeyToken=null' does not have a strong name.`

ScottPlot 4 was signed.

**ScottPlot Version:** 5.0.18

**Comment by swharden** [comment]
Thanks for reporting this @mlessmann! I'll do this for the next release 

* https://github.com/dotnet/runtime/blob/main/docs%2Fproject%2Fstrong-name-signing.md
* https://learn.microsoft.com/en-us/dotnet/standard/assembly/sign-strong-name

**Comment by swharden** [comment]
Following-up, it looks like the only action that needs to be taken is adding the following to each csproj file:

```cs
 <SignAssembly>True</SignAssembly>
```

ScottPlot 4 used to use a key for signing assemblies. The key was used for DLL identification only, not for security since the key is public: https://github.com/ScottPlot/ScottPlot/blob/main/src/ScottPlot4/ScottPlot/Key.snk (security is supported by the fact 2FA is required to upload packages to NuGet)

**Comment by swharden** [comment]
Related, it looks like source link and deterministic builds aren't working properly for ScottPlot 5. I'll keep this issue open to ensure this issue is resolved!

![image](https://github.com/ScottPlot/ScottPlot/assets/4165489/a0174e56-8149-4074-a988-0bd16d022efd)

**Comment by mlessmann** [comment]
In 5.0.19 the assemblies are still not signed. That's because adding `<SignAssembly>True</SignAssembly>` by itself is not enough. You also need to specify a key file with `<AssemblyOriginatorKeyFile>Key.snk</AssemblyOriginatorKeyFile>`.

In SP4 each project had their own key file. As I understand it it's standard practice to use the same key for all assemblies of a project.

I've submitted xxx to fix this. Unfortunately ScottPlot.Eto and ScottPlot.WPF have dependencies which are not signed either. This creates a build warning. Therefore I had to remove singing for these two projects.
